src/components/HomePage/useTransitionCards.ts

```diff
@@ -0,0 +1,15 @@
+import { useNavigate } from 'react-router-dom';
+
+export const useTransitionCards = () => {
+    const navigate = useNavigate();
+
+    const handleCardClick = (path?:string) => {
+        if (path) {
+            navigate(path);
+        }
+    }
```

Comment:
```suggestion
        const handleCardClick = useCallback((path?: string) => {
        if (!path) {
            return
        }

        navigate(path);
    }, [navigate]);
```

1. Добавить правило форматтера или линтера, чтобы ставил после двоеточия пробел path?:*пробел*string
2. Если path нет, то сразу выход. Инвертированная логика легче читается и позволяет уменьшить вложенность.
3. Обернуть в useCallback, ведь мы передаём эту функцию в компонент. Если в компоненте будет изменение стейта и ререндер, то нам эту функцию можно не пересоздавать заново.
п.3 является опциональным, знаю что есть команды, где наоборот за useCallback тут будут доёбываться